analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundantNullCheck.CSharp9.Fixed.cs

```diff
@@ -23,6 +23,8 @@
 // parenthesized pattern
 result = n is ("a" or "b" or "c"); // Fixed
 
+if (n is null && apple != null && m is null && apple is ("Sweet", "Red")) { } // FN
```

Comment:
I'll open a ticket for this, as I want to keep the timebox and this is needs a non-trivial fix - basically it should model the logical expression and for each variable inside, keep track of the verifications it has. It would be a big change.

Reply:
https://github.com/SonarSource/sonar-dotnet/issues/4645